bootstrap/kubeadm/api/v1alpha4/kubeadmconfig_webhook.go

```diff
@@ -115,8 +133,50 @@ func (c *KubeadmConfigSpec) validate(name string) error {
 		knownPaths[file.Path] = struct{}{}
 	}
 
-	if len(allErrs) == 0 {
-		return nil
+	return allErrs
+}
+
+func (c *KubeadmConfigSpec) validateIgnition() field.ErrorList {
+	var allErrs field.ErrorList
+
+	if c.Ignition != nil && c.Format != Ignition {
+		allErrs = append(allErrs, field.Invalid(
+			field.NewPath("spec", "format"), c.Format, fmt.Sprintf("must be set to %q if spec.ignition is set", Ignition)))
 	}
-	return apierrors.NewInvalid(GroupVersion.WithKind("KubeadmConfig").GroupKind(), name, allErrs)
+
+	if c.Format == Ignition && c.Ignition == nil {
+		allErrs = append(allErrs, field.Invalid(
+			field.NewPath("spec", "ignition"), c.Ignition, fmt.Sprintf("must be set if spec.format is set to %q", Ignition)))
+	}
+
+	if c.Format != Ignition {
+		return allErrs
+	}
+
+	detail := fmt.Sprintf("is not supported when spec.format is set to %q", Ignition)
+
+	for i, user := range c.Users {
+		if user.Inactive != nil && *user.Inactive {
+			allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "users", fmt.Sprintf("%d", i), "inactive"),
+				detail))
+		}
+	}
+
+	if c.UseExperimentalRetryJoin {
+		allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "useExperimentalRetryJoin"), detail))
+	}
```

Comment:
Why are we validating fields that are outside of the scope of Ignition?

Reply:
Because Ignition implementation right now does not support this field, which is shared between cloud-init and Ignition.

Reply:
@vincepri ?